### PR TITLE
Ensure reliabile loading of chapter progress on chapters page

### LINKED ISSUE #468: Live site having trouble loading ch2 again from ch-1 outro
going to try to add an isLoading switch on the chapter button of the chapters page to test.

### PR DESCRIPTION
**Pull request by benalleng**
Closes #468 

We have been consistently running into account progress errors, after some searching i found some minor logic errors that have been causing some problems with saving and initializing progress

*Test in an incognito window to fake first time user
[preview](https://saving-satoshi-git-fork-benalleng-outro-ch-3b878d-savingsatoshi.vercel.app/en/chapters/chapter-1/outro-1?dev=true)

### CODE REVIEW

**Comment by benalleng** [comment]
@ecurrencyhodler I think i found the problems with our account progress, there were some odd state changes and in combination with our local -> account -> local -> account progress was easily lost.

Before I get any real code review I would like you and I to go though and really do some deep QA on this system to try and break it in every way a user could

**Comment by ecurrencyhodler** [comment]
Amazing work! 

I would love to break it. haha. Do you want to do it on prod on in this pr?

**Comment by benalleng** [comment]
in the most up-to-date preview of this PR [here](https://saving-satoshi-git-fork-benalleng-outro-ch-3b878d-savingsatoshi.vercel.app/). At the moment prod had issues again once we implemented backwards compatibility

**Comment by ecurrencyhodler** [comment]
Okay so there's a flicker on the top right of the screen for the lock. Not a huge deal.
https://github.com/saving-satoshi/saving-satoshi/assets/32662508/5d6afe60-a406-4e67-ae28-66c564ed6b73

In this video I stop just short of completing challenge 1 by not clicking "next". But then when I go back to the chapters page I have to do redo parts 1 and 2 again. IMO it might be a lot of work to track successful completions of individual parts inside each challenge. But did want to flag it. 
https://github.com/saving-satoshi/saving-satoshi/assets/32662508/2047af90-1bd0-4549-89f8-68aaa1c18afe

**Comment by benalleng** [comment]
> Okay so there's a flicker on the top right of the screen for the lock. Not a huge deal.
Going to pick up a new PR for this as it is a little bit separate from this logic

**Comment by shaavan** [comment]
@benalleng 

I was reviewing the code, and I noticed that some parts of the code are identical to the older PR that are merged now (#436, #450)

Since these PRs are already merged, we can safely remove these changes from this PR. Also, it will make the code cleaner and easier to reason with.

**Review by ecurrencyhodler — APPROVED** [review]
tACK! Logged in with a saved account that completed ch.2. Went to Ch.1 outro-1 and ch.2 was still unlocked. Same with outro-2.

I jumped around from ch.2->ch.1 in a variety of ways. Also ch.1-Ch.2. Didn't run into any issues.

File: providers/ProgressProvider.tsx [hunk 1]
```diff
@@ -30,12 +30,14 @@ export default function ProgressProvider({
...
   )
   const [isLoading, setIsLoading] = useState(true)
-  const [savePending, setSavePending] = useState(false)
 
   const init = async () => {
     try {
       setIsLoading(true)
-      const progress = await getProgress()
+      let progress = await getProgress()
+      if (progress === 'CH1INT1') {
```
Comment (afterburn) [inline providers/ProgressProvider.tsx:38]: Nitpick but could we get `CH1INT1` from `lib/progress` instead?

**Review by afterburn — CHANGES_REQUESTED** [review]
Thanks @benalleng, requested a small change but otherwise code looks good.

File: providers/ProgressProvider.tsx (hunk 1 above)
Comment (afterburn) [reply providers/ProgressProvider.tsx:38]: Yes that, or export a new variable, as long as we can edit from the single file `lib/progress.ts` as opposed to going through the codebase to find all occurences of `CH1INT1` if we ever change it.

**Review by shaavan — APPROVED** [review]
ACK 🚀

Great work, @benalleng! No more loading problems!

**Pull request merged**